polylogue/site/builder.py

```
@@ -770,10 +866,23 @@ def _generate_conversation_pages(self, conversations: list[ConversationIndex]) -
 
-            # Skip if already exists (incremental support)
-            if page_path.exists():
-                generated += 1
-                continue
+            # Incremental: skip if page exists and is newer than conversation update
+            if incremental and page_path.exists():
+                if conv_idx.updated_at:
+                    try:
+                        from datetime import datetime
+
+                        file_mtime = datetime.fromtimestamp(page_path.stat().st_mtime)
```

Ruff (0.15.9)

[warning] `datetime.datetime.fromtimestamp()` called without a `tz` argument

(DTZ006)